lightning-invoice/src/utils.rs

```diff
@@ -353,6 +375,8 @@ where
 	if let Some(amt) = amt_msat {
 		invoice = invoice.amount_milli_satoshis(amt);
 	}
+
+	let route_hints = filter_channels(channels, amt_msat, &logger);
```

Comment:
Some changes in this method seem unrelated

Reply:
I need the `filter_channels` logging to come after the logging above, the latter of which needs `payment_hash`.